gitbutler-ui/src/lib/components/AISettings.svelte

```diff
@@ -0,0 +1,239 @@
+<script lang="ts">
+	import Select from './Select.svelte';
+	import SelectItem from './SelectItem.svelte';
+	import TextBox from './TextBox.svelte';
+	import { GitConfig } from '$lib/backend/gitConfig';
+	import {
+		AnthropicModel,
+		KeyOption,
+		ModelKind,
+		OpenAIModel,
+		SummarizerSettings
+	} from '$lib/backend/summarizerSettings';
+	import RadioButton from '$lib/components/RadioButton.svelte';
+	import SectionCard from '$lib/components/SectionCard.svelte';
+
+	export let summarizerSettings = new SummarizerSettings(new GitConfig());
+
+	let modelKind: ModelKind | undefined;
+	summarizerSettings.getModelKind().then((persistedModelKind) => (modelKind = persistedModelKind));
+	$: if (modelKind) summarizerSettings.setModelKind(modelKind);
+	$: if (form && modelKind) form.modelKind.value = modelKind;
+
+	let keyOption: { name: string; value: KeyOption } | undefined;
+	summarizerSettings
+		.getKeyOption()
+		.then(
+			(persistedKeyOption) =>
+				(keyOption = keyOptions.find((option) => option.value == persistedKeyOption))
+		);
```

Comment:
How about moving these into an `onMount(async ()=> {})` so you can use `await` instead of `then`?

Reply:
I like this suggestion, will do 👍